dlpy/tests/test_model.py

```diff
@@ -1038,7 +1041,822 @@ def test_mobilenetv2(self):
+        model1.add(InputLayer(3, 224, 224))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Dense(16))
+        model1.add(OutputLayer(act='softmax', n=2))
+
+        tensorboard = TensorBoard(model1, log_dir)
+        r = model1.fit(data='eee', inputs='_image_', target='_label_', lr=0.0001, tensorboard=tensorboard)
+        
+        epochs = model1.training_history.Epoch.values
+        lr = model1.training_history.LearningRate.values
+        loss = model1.training_history.Loss.values
+        error = model1.training_history.FitError.values
+        self.assertTrue(len(epochs) == 5)
+
+        # Check number of event files created (1 for each call to fit)
+        self.assertTrue(len(os.listdir(log_dir +'Simple_CNN1' + '/learning_rate/')) == 1)
+        self.assertTrue(len(os.listdir(log_dir +'Simple_CNN1' + '/loss/')) == 1)
+        self.assertTrue(len(os.listdir(log_dir +'Simple_CNN1' + '/error/')) == 1)
+
+        # Check lr
+        count = 0
+        tfevent_file = os.listdir(log_dir +'Simple_CNN1' + '/learning_rate/')
+        for file in tfevent_file:
+            serialized_examples = tf.data.TFRecordDataset(log_dir +'Simple_CNN1' + '/learning_rate/' + file)
+            for serialized_example in serialized_examples:
+                event = event_pb2.Event.FromString(serialized_example.numpy())
+                for value in event.summary.value:
+                    t = tf.make_ndarray(value.tensor)
+                    self.assertAlmostEqual(t, lr[count], places=3)
+                    count += 1
+        self.assertEquals(count, len(epochs))
+
+        # Check loss
+        count = 0
+        tfevent_file = os.listdir(log_dir +'Simple_CNN1' + '/loss/')
+        for file in tfevent_file:
+            serialized_examples = tf.data.TFRecordDataset(log_dir +'Simple_CNN1' + '/loss/' + file)
+            for serialized_example in serialized_examples:
+                event = event_pb2.Event.FromString(serialized_example.numpy())
+                for value in event.summary.value:
+                    t = tf.make_ndarray(value.tensor)
+                    self.assertAlmostEqual(t, loss[count], places=3)
+                    count += 1
+        self.assertEquals(count, len(epochs))
+        
+        # Check error
+        count = 0
+        tfevent_file = os.listdir(log_dir +'Simple_CNN1' + '/error/')
+        for file in tfevent_file:
+            serialized_examples = tf.data.TFRecordDataset(log_dir +'Simple_CNN1' + '/error/' + file)
+            for serialized_example in serialized_examples:
+                event = event_pb2.Event.FromString(serialized_example.numpy())
+                for value in event.summary.value:
+                    t = tf.make_ndarray(value.tensor)
+                    self.assertAlmostEqual(t, error[count], places=3)
+                    count += 1
+        self.assertEquals(count, len(epochs))
```

Comment:
does this work after completing the training? or in real time?

Reply:
Not sure I follow here so please correct me if I'm misunderstanding. 

The logging occurs in real time. So you will see TensorBoard update at whatever the refresh rate is (I think 12sec by default) with the newly output values from fit_tensorboard().